spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -356,7 +359,7 @@ object Engine {
       val patternLocalStar = "local\\[\\*\\]".r
       master match {
         case patternLocalN(n) => Some(1, n.toInt)
-        case patternLocalStar(_*) => Some(1, getCoreNumberFromEnv)
+        case patternLocalStar(_*) => Some(1, getNumMachineCores)
```

Comment:
This changes the original logic. In the original code, local[*] will use DL_CORE_NUMBER cores if it is set, then all the cores on the machine. Now, it will ignore the DL_CORE_NUMBER setting.

I think the change is reasonable. But it breaks some unit tests in python if the machine cores number is not 4 or 8(e.g. on some server). 

After some dig, I find that the root cause is even we specify the master to local[4] in python test file pyspark/test/simple_integration_test.py line:30, the master config value read in scala is local[*]...